PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx

```diff
@@ -71,314 +108,366 @@ static constexpr std::string_view TrackVariableNames[kLAST_TrackVariable] = {"Pt
+    registry.fill(HIST(AR::RecoSim[rs]) + HIST("TrackControl/") + HIST(AR::BeforeAfter[ba]) + HIST(AR::TrackVariableNames[AR::kTPCCHI2]),
+                  track.tpcChi2NCl());
+    registry.fill(HIST(AR::RecoSim[rs]) + HIST("TrackControl/") + HIST(AR::BeforeAfter[ba]) + HIST(AR::TrackVariableNames[AR::kITSCLUSTERS]),
+                  track.itsNCls());
+  }
 
-    // cut vx
-    if (cfgEvent.at(AR::kVX).value.at(5) > 0 &&
-        (collision.posX() < cfgEvent.at(AR::kVX).value.at(3) || collision.posX() > cfgEvent.at(AR::kVX).value.at(4))) {
-      SurviveCut = false;
-    }
-    // cut vy
-    if (cfgEvent.at(AR::kVY).value.at(5) > 0 &&
-        (collision.posY() < cfgEvent.at(AR::kVY).value.at(3) || collision.posY() > cfgEvent.at(AR::kVY).value.at(4))) {
-      SurviveCut = false;
-    }
-    // cut vz
-    if (cfgEvent.at(AR::kVZ).value.at(5) > 0 &&
-        (collision.posZ() < cfgEvent.at(AR::kVZ).value.at(3) || collision.posZ() > cfgEvent.at(AR::kVZ).value.at(4))) {
-      SurviveCut = false;
-    }
-    // cut vabs
-    if (cfgEvent.at(AR::kVABS).value.at(5) > 0 &&
-        (std::sqrt(std::pow(collision.posX(), 2) + std::pow(collision.posY(), 2) + std::pow(collision.posZ(), 2)) < cfgEvent.at(AR::kVABS).value.at(3) || std::sqrt(std::pow(collision.posX(), 2) + std::pow(collision.posY(), 2) + std::pow(collision.posZ(), 2)) > cfgEvent.at(AR::kVABS).value.at(4))) {
-      SurviveCut = false;
-    }
-    // cut centrality
-    if (cfgEvent.at(AR::kCEN).value.at(5) > 0 &&
-        (collision.centRun2V0M() < cfgEvent.at(AR::kCEN).value.at(3) || collision.centRun2V0M() > cfgEvent.at(AR::kCEN).value.at(4))) {
-      SurviveCut = false;
-    }
-    // cut multiplicity (qvector)
-    if (cfgEvent.at(AR::kMULQ).value.at(5) > 0 &&
-        (collision.size() < cfgEvent.at(AR::kMULQ).value.at(3) || collision.size() > cfgEvent.at(AR::kMULQ).value.at(4))) {
-      SurviveCut = false;
-    }
-    // cut multiplicity (weights)
-    if (cfgEvent.at(AR::kMULW).value.at(5) > 0 &&
-        (collision.size() < cfgEvent.at(AR::kMULW).value.at(3) || collision.size() > cfgEvent.at(AR::kMULW).value.at(4))) {
-      SurviveCut = false;
+  // function for checking if collision survives event cuts
+  template <typename CollisionObject, typename TrackObject>
+  bool SurviveEventCuts(CollisionObject collision, TrackObject tracks)
+  {
+    bool flag = true;
+
+    // Check if event survives event cuts, where we can get the values for the variables immediately
+    if (!AR::SurviveCut(cfgVX.value, collision.posX()) ||
+        !AR::SurviveCut(cfgVY.value, collision.posY()) ||
+        !AR::SurviveCut(cfgVZ.value, collision.posZ()) ||
+        !AR::SurviveCut(cfgVABS.value, AR::abs(collision.posX(), collision.posY(), collision.posZ())) ||
+        !AR::SurviveCut(cfgCEN.value, collision.centRun2V0M()) ||
+        !AR::SurviveCut(cfgMULNC.value, collision.numContrib()) ||
+        !AR::SurviveCut(cfgMULTPC.value, collision.multTPC())) {
+      flag = false;
```

Comment:
```suggestion
      return false;
```

Reply:
With this, you don't need `if` in line 360

Reply:
Fixed.